diff -Naur ../gkrellmms-g2/Makefile ./Makefile
--- ../gkrellmms-g2/Makefile	2010-11-03 21:30:41.000000000 -0400
+++ ./Makefile	2010-11-03 22:27:02.000000000 -0400
@@ -8,8 +8,8 @@
 IMLIB_INCLUDE ?= 
 IMLIB_LIB ?= 
 
-XMMS_INCLUDE ?= `pkg-config audacious --cflags`
-XMMS_LIB ?= `pkg-config audacious --libs`
+XMMS_INCLUDE ?= `pkg-config audclient --cflags`
+XMMS_LIB ?= `pkg-config audclient --libs`
 
 DBUS_INCLUDE ?= `pkg-config dbus-1 --cflags`
 DBUS_LIB ?= `pkg-config dbus-1 --libs`
diff -Naur ../gkrellmms-g2/playlist.h ./playlist.h
--- ../gkrellmms-g2/playlist.h	2010-11-03 21:30:33.000000000 -0400
+++ ./playlist.h	2010-11-03 21:35:40.000000000 -0400
@@ -24,8 +24,6 @@
 #include <stdio.h>
 #include <time.h>
 
-#include <audacious/auddrct.h>
-
 #include "gkrellmms.h"
 
 /* init the playlist */
