diff -Naur gkrellm-reminder-2.0.0/Makefile gkrellm-reminder-2.0.0-r1/Makefile
--- gkrellm-reminder-2.0.0/Makefile	2002-05-01 13:39:37.000000000 -0400
+++ gkrellm-reminder-2.0.0-r1/Makefile	2010-12-26 02:43:55.000000000 -0500
@@ -1,18 +1,18 @@
 GTK_INCLUDE = `pkg-config gtk+-2.0 --cflags`
 GTK_LIB = `pkg-config gtk+-2.0 --libs`
 
-CFLAGS = -O2 -Wall -fPIC $(GTK_INCLUDE) -I/usr/pkg/include
+CFLAGS = -O2 -Wall -fPIC $(GTK_INCLUDE)
 
 LIBS = $(GTK_LIB)
 
-LFLAGS = -shared -L/usr/pkg/lib
+LFLAGS = -shared
 
 CC = gcc
 
 all: reminder.so
 
 reminder.so: reminder.c calendar.xpm
-	$(CC) $(CFLAGS) -o reminder.so $(LFLAGS) $(LIBS) reminder.c
+	$(CC) $(CFLAGS) -o reminder.so $(LFLAGS) $(LDFLAGS) $(LIBS) reminder.c
 
 clean:
 	rm -f *.o core *.so
diff -Naur gkrellm-reminder-2.0.0/reminder.c gkrellm-reminder-2.0.0-r1/reminder.c
--- gkrellm-reminder-2.0.0/reminder.c	2002-12-04 00:29:31.000000000 -0500
+++ gkrellm-reminder-2.0.0-r1/reminder.c	2010-12-26 02:42:47.000000000 -0500
@@ -87,7 +87,7 @@
 
 struct event_stored {
   gchar *name;
-  guint id;
+  gulong id;
   gint days;
   gint occurs;
   time_t start;
@@ -99,7 +99,7 @@
 
 struct event_today {
   gchar *name;
-  guint id;
+  gulong id;
   time_t occurs;
   gboolean last;
 
@@ -107,7 +107,7 @@
 };
 
 struct id_list {
-  guint id;
+  gulong id;
 
   struct id_list *next;
 };
@@ -333,7 +333,7 @@
   //    strcpy( current->name, buffer );
       current->name = g_strdup(buffer);
 
-      if( fscanf( fp, "%u %d %d %ld %ld %ld\n", &current->id, &current->days,
+      if( fscanf( fp, "%lu %d %d %ld %ld %ld\n", &current->id, &current->days,
           &current->occurs, &current->start, &current->end,
           &current->last_displayed ) != 6 )
 	{
@@ -431,7 +431,7 @@
   current = head_stored;
   while( current )
     {
-      fprintf( fp, "%s\n%u %d %d %ld %ld %ld\n", current->name, current->id, current->days,
+      fprintf( fp, "%s\n%lu %d %d %ld %ld %ld\n", current->name, current->id, current->days,
 	       current->occurs, current->start, current->end, current->last_displayed );
 
       current = current->next;
@@ -442,7 +442,7 @@
 }
 
 static void
-reminder_remove_event_today( guint id )
+reminder_remove_event_today( gulong id )
 {
   struct event_today *current, *temp;
 
@@ -482,7 +482,7 @@
 }
 
 static gboolean
-reminder_remove_event_stored( struct event_stored **head_list, guint id )
+reminder_remove_event_stored( struct event_stored **head_list, gulong id )
 {
   struct event_stored *current, *temp;
 
@@ -529,7 +529,7 @@
 }
 
 static struct event_stored *
-reminder_find_event_stored( struct event_stored *head_list, guint id )
+reminder_find_event_stored( struct event_stored *head_list, gulong id )
 {
   struct event_stored *current;
 
@@ -1003,8 +1003,8 @@
 static void
 cb_set_days( GtkWidget *window, gpointer data )
 {
-  gint weekdays = (gint) data & 1;
-  gint weekends = (gint) data & 2;
+  gint weekdays = (long) data & 1;
+  gint weekends = (long) data & 2;
 
   gtk_toggle_button_set_active( GTK_TOGGLE_BUTTON( check_sun ), weekends );
   gtk_toggle_button_set_active( GTK_TOGGLE_BUTTON( check_sat ), weekends );
@@ -1340,7 +1340,7 @@
 static void
 cb_add( GtkWidget *window, gpointer data )
 {
-  guint id;
+  gulong id;
   struct event_stored *current, *tail;
 
   id = (guint) time( NULL );
@@ -1383,12 +1383,12 @@
 cb_remove( GtkWidget *window, gpointer data )
 {
   struct id_list *current;
-  guint id;
+  gulong id;
 
   if( list_main_row_selected == -1 )
     return;
 
-  id = (guint) gtk_clist_get_row_data( GTK_CLIST( list_main ), list_main_row_selected );
+  id = (gulong) gtk_clist_get_row_data( GTK_CLIST( list_main ), list_main_row_selected );
 
   /* Try to remove event from temp list. If not, add to to-be-deleted list */
   if( !reminder_remove_event_stored( &head_temp, id ) )
@@ -1661,13 +1661,13 @@
 
   struct event_stored *es1, *es2;
 
-  es1 = reminder_find_event_stored( head_stored, (guint) row1->data );
+  es1 = reminder_find_event_stored( head_stored, (gulong) row1->data );
   if( !es1 )
-    es1 = reminder_find_event_stored( head_temp, (guint) row1->data );
+    es1 = reminder_find_event_stored( head_temp, (gulong) row1->data );
 
-  es2 = reminder_find_event_stored( head_stored, (guint) row2->data );
+  es2 = reminder_find_event_stored( head_stored, (gulong) row2->data );
   if( !es2 )
-    es2 = reminder_find_event_stored( head_temp, (guint) row2->data );
+    es2 = reminder_find_event_stored( head_temp, (gulong) row2->data );
 
   if( es1 && es2 )
     {
@@ -1692,13 +1692,13 @@
 
   struct event_stored *es1, *es2;
 
-  es1 = reminder_find_event_stored( head_stored, (guint) row1->data );
+  es1 = reminder_find_event_stored( head_stored, (gulong) row1->data );
   if( !es1 )
-    es1 = reminder_find_event_stored( head_temp, (guint) row1->data );
+    es1 = reminder_find_event_stored( head_temp, (gulong) row1->data );
 
-  es2 = reminder_find_event_stored( head_stored, (guint) row2->data );
+  es2 = reminder_find_event_stored( head_stored, (gulong) row2->data );
   if( !es2 )
-    es2 = reminder_find_event_stored( head_temp, (guint) row2->data );
+    es2 = reminder_find_event_stored( head_temp, (gulong) row2->data );
 
   if( es1 && es2 )
     return( ( ( es1->start - TIMEZONE_DIFF ) % SECS_PER_DAY ) -
@@ -1715,13 +1715,13 @@
 
   struct event_stored *es1, *es2;
 
-  es1 = reminder_find_event_stored( head_stored, (guint) row1->data );
+  es1 = reminder_find_event_stored( head_stored, (gulong) row1->data );
   if( !es1 )
-    es1 = reminder_find_event_stored( head_temp, (guint) row1->data );
+    es1 = reminder_find_event_stored( head_temp, (gulong) row1->data );
 
-  es2 = reminder_find_event_stored( head_stored, (guint) row2->data );
+  es2 = reminder_find_event_stored( head_stored, (gulong) row2->data );
   if( !es2 )
-    es2 = reminder_find_event_stored( head_temp, (guint) row2->data );
+    es2 = reminder_find_event_stored( head_temp, (gulong) row2->data );
 
   if( es1 && es2 )
     return es1->start - es2->start;
@@ -1737,13 +1737,13 @@
 
   struct event_stored *es1, *es2;
 
-  es1 = reminder_find_event_stored( head_stored, (guint) row1->data );
+  es1 = reminder_find_event_stored( head_stored, (gulong) row1->data );
   if( !es1 )
-    es1 = reminder_find_event_stored( head_temp, (guint) row1->data );
+    es1 = reminder_find_event_stored( head_temp, (gulong) row1->data );
 
-  es2 = reminder_find_event_stored( head_stored, (guint) row2->data );
+  es2 = reminder_find_event_stored( head_stored, (gulong) row2->data );
   if( !es2 )
-    es2 = reminder_find_event_stored( head_temp, (guint) row2->data );
+    es2 = reminder_find_event_stored( head_temp, (gulong) row2->data );
 
   if( es1 && es2 )
     return es1->end - es2->end;
@@ -2884,7 +2884,7 @@
 static void
 reminder_window_never( GtkWidget *window, gpointer data )
 {
-  guint id = (guint) data;
+  gulong id = (gulong) data;
 
   /* delete event from today */
   num_active--;
@@ -2914,7 +2914,7 @@
 static void
 reminder_window_later( GtkWidget *window, gpointer data )
 {
-  guint id = (guint) data;
+  gulong id = (gulong) data;
   struct event_stored *new, *old;
 
   /* delete event from today */
@@ -2991,7 +2991,7 @@
 static void
 reminder_window_dismiss( GtkWidget *window, gpointer data )
 {
-  guint id = (guint) data;
+  gulong id = (gulong) data;
 
   /* delete event from today */
   num_active--;
@@ -3181,12 +3181,12 @@
 
   ts_old = ts = reminder_text_decal->text_style;
 
-  w = gdk_string_width( ts.font, buf );
+  w = gdk_string_width( gdk_font_from_description(ts.font), buf );
 
   if( w > reminder_text_decal->w )
     {
       ts.font = gkrellm_meter_alt_textstyle( style_id )->font;
-      w = gdk_string_width( ts.font, buf );
+      w = gdk_string_width( gdk_font_from_description(ts.font), buf );
     }
 
   style = gkrellm_meter_style( style_id );
