diff '--color=auto' -u --recursive a/inews/clientlib.c b/inews/clientlib.c
--- a/inews/clientlib.c	2025-12-19 00:04:17.904180177 -0500
+++ b/inews/clientlib.c	2025-12-19 01:11:13.621279245 -0500
@@ -72,9 +72,7 @@
  *	Side effects:	None.
  */
 
-char *
-getserverbyfile(file)
-char	*file;
+char *getserverbyfile(char *file)
 {
 	register FILE	*fp;
 	register char	*cp;
@@ -135,8 +133,7 @@
  *			for reading and writing to server.
  */
 
-int server_init(machine)
-char	*machine;
+int server_init(char *machine)
 {
 	int	sockt_rd, sockt_wr;
 	char	line[256];
@@ -196,14 +193,13 @@
  *	Errors:		Printed via perror.
  */
 
-int get_tcp_socket(machine)
-char	*machine;
+int get_tcp_socket(char *machine)
 {
 	int	s;
 	struct	sockaddr_in sin;
 #ifndef EXCELAN
-	struct	servent *getservbyname(), *sp;
-	struct	hostent *gethostbyname(), *hp;
+	struct	servent *getservbyname(const char *, const char *), *sp;
+	struct	hostent *gethostbyname(const char *), *hp;
 #ifdef h_addr
 	int	x = 0;
 	register char **cp;
@@ -345,8 +341,7 @@
  *	Errors:		Printed via nerror.
  */
 
-int get_dnet_socket(machine)
-char	*machine;
+int get_dnet_socket(char *machine)
 {
 	int	s, area, node;
 	struct	sockaddr_dn sdn;
@@ -428,9 +423,7 @@
  *	Side effects:	None.
  */
 
-int handle_server_response(response, server)
-int	response;
-char	*server;
+int handle_server_response(int response, char *server)
 {
     switch (response) {
 	case OK_NOPOST:		/* fall through */
@@ -478,9 +471,7 @@
  *			fflush.
  */
 
-void
-put_server(string)
-char *string;
+void put_server(char *string)
 {
 #ifdef DEBUG
 	fprintf(stderr, ">>> %s\n", string);
@@ -503,9 +494,7 @@
  *	Side effects:	Talks to server, changes contents of "string".
  */
 
-int get_server(string, size)
-char	*string;
-int	size;
+int get_server(char *string, int size)
 {
 	register char *cp;
 
@@ -537,8 +526,7 @@
  *			after this routine is called.
  */
 
-void
-close_server()
+void close_server()
 {
 	char	ser_line[256];
 
diff '--color=auto' -u --recursive a/inews/clientlib.h b/inews/clientlib.h
--- a/inews/clientlib.h	2025-12-19 00:04:17.904180177 -0500
+++ b/inews/clientlib.h	2025-12-19 00:51:48.618334135 -0500
@@ -4,10 +4,10 @@
  * @(#)clientlib.h	1.1	(Berkeley) 1/9/88
  */
 
-extern	char	*getserverbyfile();
-extern	int	server_init();
-extern	void	put_server();
-extern	int	get_server();
+extern	char	*getserverbyfile(char *);
+extern	int	server_init(char *);
+extern	void	put_server(char *);
+extern	int	get_server(char *string, int size);
 extern	void	close_server();
 
 extern	int	get_tcp_socket(char *machine);
diff '--color=auto' -u --recursive a/inews/inews.c b/inews/inews.c
--- a/inews/inews.c	2025-12-19 00:04:17.904180177 -0500
+++ b/inews/inews.c	2025-12-19 01:07:11.069504697 -0500
@@ -52,15 +52,13 @@
 
 char	host_name[256];
 
-int main(argc, argv)
-int	argc;
-char	*argv[];
+int main(int argc, char *argv[])
 {
 	char	line[NNTP_STRLEN], s[NNTP_STRLEN];
 	int	seen_fromline, in_header, seen_header;
 	int	response;
 	char	*server;
-	char	*getserverbyfile();
+	char	*getserverbyfile(char *);
 	register char	*cp;
 
 	++argv;
@@ -335,11 +333,9 @@
  *	Side effects:	None.
  */
 
-int strneql(a, b, n)
-register char *a, *b;
-int	n;
+int strneql(register char *a, register char *b, int	n)
 {
-	char	lower();
+	char	lower(char c);
 
 	while (n && lower(*a) == lower(*b)) {
 		if (*a == '\0')
@@ -368,8 +364,7 @@
  *	Side effects:	None.
  */
 
-char lower(c)
-register char c;
+char lower(register char c)
 {
 	if (isascii(c) && isupper(c))
 		c = c - 'A' + 'a';
@@ -388,8 +383,7 @@
  *
  */
 
-int valid_header(h)
-register char *h;
+int valid_header(register char *h)
 {
   char *colon, *space;
 
diff '--color=auto' -u --recursive a/uulib/uuencode.c b/uulib/uuencode.c
--- a/uulib/uuencode.c	2002-04-02 05:04:52.000000000 -0500
+++ b/uulib/uuencode.c	2025-12-19 01:02:01.264788443 -0500
@@ -931,7 +931,7 @@
 		 eolstring);
       }
 
-      fprintf (outfile, "=ypart begin=%d end=%d%s",
+      fprintf (outfile, "=ypart begin=%ld end=%ld%s",
 	       (partno-1)*linperfile*128+1,
 	       (partno*linperfile*128) < progress.totsize ? 
 	       (partno*linperfile*128) : progress.totsize,
@@ -997,14 +997,14 @@
   }
   else if (encoding == YENC_ENCODED) {
     if (numparts != 1) {
-      fprintf (outfile, "=yend size=%d part=%d pcrc32=%08lx",
+      fprintf (outfile, "=yend size=%ld part=%d pcrc32=%08lx",
 	       (partno*linperfile*128) < progress.totsize ? 
 	       linperfile*128 : (progress.totsize-(partno-1)*linperfile*128),
 	       partno,
 	       pcrc);
     }
     else {
-      fprintf (outfile, "=yend size=%d",
+      fprintf (outfile, "=yend size=%ld",
 	       progress.totsize);
     }
     if (feof (theifile))
@@ -1405,7 +1405,7 @@
 		   eolstring);
 	}
 
-	fprintf (outfile, "=ypart begin=%d end=%d%s",
+	fprintf (outfile, "=ypart begin=%ld end=%ld%s",
 		 (part-1)*linperfile*128+1,
 		 (part*linperfile*128) < progress.totsize ? 
 		 (part*linperfile*128) : progress.totsize,
@@ -1451,14 +1451,14 @@
     }
     else if (encoding == YENC_ENCODED) {
       if (numparts != 1) {
-	fprintf (outfile, "=yend size=%d part=%d pcrc32=%08lx",
+	fprintf (outfile, "=yend size=%ld part=%d pcrc32=%08lx",
 		 (part*linperfile*128) < progress.totsize ? 
 		 linperfile*128 : (progress.totsize-(part-1)*linperfile*128),
 		 part,
 		 pcrc);
       }
       else {
-	fprintf (outfile, "=yend size=%d",
+	fprintf (outfile, "=yend size=%ld",
 		 progress.totsize);
       }
       if (feof (theifile))
diff '--color=auto' -u --recursive a/uulib/uunconc.c b/uulib/uunconc.c
--- a/uulib/uunconc.c	2025-12-19 00:03:23.082244867 -0500
+++ b/uulib/uunconc.c	2025-12-19 01:27:56.495341720 -0500
@@ -1583,7 +1583,10 @@
     r[0] = fgetc (datain);
     hb   = (int) r[0] + 22;
     fseek (datain, (int) r[0] + 12, SEEK_SET);
-    fread (r, 1, 8, datain);
+    if (fread (r, 1, 8, datain) < 8) {
+        perror("fread");
+        exit(1);
+    }
 
     dsize = (((long) 1 << 24) * (long) r[0]) +
             (((long) 1 << 16) * (long) r[1]) +
